.github/workflows/pypi.yml

```diff
@@ -17,9 +17,34 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  check_changes:
+    runs-on: ubuntu-latest
+    outputs:
+      should_run: ${{ steps.filter.outputs.any_changed }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          filters: |
+            any_changed:
+              - '.github/workflows/pypi.yml'
+              - 'pyproject.toml'
+              - 'setup.py'
+              - 'pytensor/_version.py'
+              - 'pytensor/scan_perform.pyx'
+              - 'pytensor/scan_perform_ext.py'
+
   # The job to build precompiled pypi wheels.
   make_sdist:
     name: Make SDist
+    needs: check_changes
+    # Run if it's a release, auto-release branch, or if relevant files changed on main
+    if: |
+      github.event_name == 'release' || 
+      github.ref == 'refs/heads/auto-release' || 
```

Comment:
I believe we should delete all references to `auto-release` AFAIK it predates PyTensor